test/passing/break_cases.ml

```diff
@@ -0,0 +1,25 @@
+let f x = function
+  | C ->
+      1
+  | D ->
+      2
```

Comment:
I would have expected `--break-case` to format this into:

```ocaml
let f x = function
  | C -> 1
  | D -> 2
```

Also I am wondering if we can do something about alignements of `->` (not in scope for this PR), e.g:

```ocaml
let f x = function
  | C (x, _) -> x
  | D        -> 2
```

Reply:
I agree it would look better, I don't know why the line breaks after the arrow but I will look into it.

Reply:
@samoht I think this is due to the way this patch interacts with the two prior modes. The current `--sparse` mode adds breaks after `->` in order to avoid the formatting of nested or-patterns which changes the indentation of `|` to indicate the nesting, which some people find odd. When nested or-patterns are not visually distinguished, there is a source of bugs for code that has a long sequence of nested or-patterns with some short body partway though. Without breaking after `->` or indicating the nesting structure with the indentation of `|`, it is very easy to visually mis-read the code, concluding that a later body covers an earlier pattern, when in fact there is a "hidden" body in between. Manually formatting code can add a break for the hidden body to avoid this, but it is harder with auto-formatting. Given the current implementation of ocamlformat in terms of the stdlib Format module, I don't think it is possible to force two non-nested boxes to either both break or both not break. Without that implementation possibility, I don't know of other formatting algorithms for code like this that avoids the hidden body danger. Other ideas?

Vertical alignment is tricky in general, at least when what appears to the left of the desired alignment point (e.g. the patterns before the `->`s) does not have a statically fixed length. To implement such alignment, I think it seems necessary to first format all the patterns, then compute their lengths and whether they broke over multiple lines, and then backtrack to format the entire list of cases. The question of whether each pattern fits on a line ought to be sensitive to the starting indentation, which isn't possible with the current design using Format I think. (Format does not support much backtracking, and doing so quickly makes the core algorithms exponential, which might be survivable, but maybe not.) Something rougher could be done instead, such as requiring the patterns to be short enough to fit in e.g. one third of the margin width in order to align them, and then just hope that the starting indentation does not exceed that in practice. I have been meaning to try that, but have not had time, and am not sure how optimistic to be about how well it might work.